platform/osx/app/MapDocument.m

```diff
@@ -457,8 +488,8 @@ - (BOOL)validateMenuItem:(NSMenuItem *)menuItem {
     if (menuItem.action == @selector(dropManyPins:)) {
         return YES;
     }
-    if (menuItem.action == @selector(removeAllPins:)) {
-        return self.mapView.annotations.count;
+    if (menuItem.action == @selector(removeAllAnnotations:)) {
+        return self.mapView.annotations.count ? YES : NO;
```

Comment:
Eh, this isn’t strictly necessary. If you prefer to be explicit, it’d be better to return `self.mapView.annotations.count > 0`.